templates/gitserver/gitserver.StatefulSet.yaml

```diff
@@ -6,29 +6,25 @@
 {{- $_ := set $envVars "ORIGIN_MAP" $global.Values.site.gitOriginMap -}}
 {{- include "collectEnv" (list $envVars .Values.cluster.gitserver.containers.gitserver.env) -}}
 
-{{- range $index, $gitserverIndex := until (required ".Values.cluster.gitserver.shards required" .Values.cluster.gitserver.shards | int) }}
-{{- if eq $index 0 | not }}
----
-{{ end -}}
-apiVersion: extensions/v1beta1
-kind: Deployment
+apiVersion: apps/v1
+kind: StatefulSet
 metadata:
   annotations:
     description: Stores clones of repositories to perform Git operations.
-  name: gitserver-{{ add1 $gitserverIndex }}
+  name: gitserver
 spec:
-  minReadySeconds: 10
-  replicas: 1
+  serviceName: gitserver
+  selector:
+    matchLabels:
+      app: gitserver
+  replicas: {{ required ".Values.cluster.gitserver.shards required" .Values.cluster.gitserver.shards | int }}
   revisionHistoryLimit: 10
-  strategy:
-    rollingUpdate:
```

Comment:
`maxSurge` and `maxUnavailable` aren't supported for stateful sets https://godoc.org/k8s.io/api/apps/v1#RollingUpdateStatefulSetStrategy